### DISCUSSION TITLE
Is it possible to create Renderable from PathShape?

### DISCUSSION THREAD

**Discussion by AlexMarikutsa** [sceneview/sceneview#132] [Q&A]
Good day! Could you help me to create Renderable from PathShape and add it to node? Like it is implemented in SCNShape from iOS. Thank you.

**Comment by ThomasGorisse** [comment]
Yes! In v4.0.0-rc.1, you can create geometry from paths and shapes. The `sceneview-core` module includes triangulation (Earcut, Delaunator) and geometry generation:

```kotlin
SceneView {
    // Built-in geometry nodes:
    CubeNode(materialInstance = material, size = Size(1f))
    SphereNode(materialInstance = material, radius = 0.5f)
    CylinderNode(materialInstance = material, radius = 0.5f, height = 1f)
    
    // For custom shapes, use MeshNode with generated geometry
    MeshNode(materialInstance = material, vertices = ..., indices = ...)
}
```

The `sceneview-core` KMP module provides `Shape`, `Path`, and `Line` geometry generators that you can use with `MeshNode` for arbitrary 2D shapes extruded into 3D.
